### PR TITLE
Added start command for running seamlessly from docker

### PR DESCRIPTION
**Pull request by JoelLarson**
This command will allow anyone with docker to easily build pegaswitch without needing the nodejs requirement on their local machine.

It will also have the added benefit of allowing pegaswitch to easily run on any platform without hassle.

Let me know if there's any feedback or improvements if this can get in!

### CODE REVIEW

File: bin/start.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+which docker
+
+if [ $? -ne 0 ]; then
+    echo "Docker must be installed to use this feature. Exiting..." >&2
```
Comment (SomeoneWeird) [inline bin/start.sh]: You don't actually exit here?

**Review by SomeoneWeird — CHANGES_REQUESTED** [review]
Should probably also add a `start-docker` command or something in package.json and add it to the readme too.

**Comment by JoelLarson** [comment]
@SomeoneWeird Thanks for the feedback! I made all the changes for it. I'm about to add a executable file for windows as well that I'm testing on another machine.

**Comment by misson20000** [comment]
Can we pass through arguments to start.js? I almost never use the defaults.

**Comment by JoelLarson** [comment]
@misson20000 I added an argument passthrough for it.

`./bin/start-docker.sh --ip 1.2.3.4` should work as expected.

File: bin/start-docker.sh [hunk 2]
```diff
@@ -24,11 +24,13 @@ if [ ! -d "node_modules" ]; then
     echo "Node modules will install on the first run"
 fi
 
+exit 0
```
Comment (SomeoneWeird) [inline bin/start-docker.sh]: I don't think you want this :)

Comment (JoelLarson) [reply bin/start-docker.sh]: Debug code... lol

**Comment by JoelLarson** [comment]
It's now removed

**Comment by JoelLarson** [comment]
I'm having some npm install issues with the version on windows so I think I'll make a pull request for that in the future. This one can be marked done if there isn't anything else lingering.

**Comment by JoelLarson** [comment]
@SomeoneWeird I have added Windows support! It is now as simple as double clicking the bat file.

File: README.md [hunk 3]
```diff
@@ -30,7 +47,7 @@ Documentation
 
 API documentation for SploitCore is automatically generated using jsdoc comments.
 
-You can find the latest version of documentation hosted [here](https://reswitched.github.io/pegaswitch/)
+You can find the latest version of documentation hosted [here](https://reswitched.github.io/PegaSwitch/)
```
Comment (misson20000) [inline README.md]: This link is case sensitive

**Comment by misson20000** [comment]
This does not seem to be passing arguments to pegaswitch properly. This is my usual command line:
```
$ sudo node start.js --disable-curses --logfile log.txt --ip 10.0.0.192 --setuid misson20000
```
When I run this,
```
$ bin/start-docker.sh --disable-curses --logfile log.txt --ip 10.0.0.192 --setuid misson20000
```
PegaSwitch still opens the curses interface and reports that
```
DNS Server started, responding with address 172.17.0.2
```

**Comment by JoelLarson** [comment]
@misson20000 I have fixed that bug now. I accidentally used single quotes instead of double quotes while I was trying to fix something else.

**Comment by meeces2911** [comment]
Also, to add to that, you might want to specify port 53 as UDP in the docker port bind.

**Comment by JoelLarson** [comment]
Thanks @meeces2911! Just fixed both problems. I'll test them when I get home tonight.

**Comment by JoelLarson** [comment]
- [x] Tested running `./bin/start-docker.bat` on Windows by double clicking.
- [x] Tested running `./bin/start-docker.bat --ip 1.2.3.4` on Windows.
- [x] Tested `./bin/start-docker.sh` runs from root path.
- [x] Tested `./bin/start-docker.sh --disable-curses --logfile log.txt --ip 10.0.0.192`
- [x] Tested `npm run-script start-docker`
- [x] Tested `npm run-script start-docker -- --disable-curses --logfile log.txt --ip 10.0.0.192`
- [x] Tested that DNS port was open (`53/udp`)
- [x] Tested that HTTP port was open (`80/tcp`)
- [] Tested that WebSocket port was open and correct (`8100/tcp`)

DNS Test
`dig @localhost -p 53 foo.example A` should confirm what `--ip` is set to.

Web Test
`curl --silent -X GET http://localhost/ | grep -q -i 'PegaSwitch' && echo 'Serving PegaSwitch.' || echo 'PegaSwitch is not being served correctly.'`

I'm not sure how to test port 8100 (the websocket) since I think it mounts only when the program is running and I do not have a Switch with a compatible OS.

**Comment by JoelLarson** [comment]
@SomeoneWeird Is anything left for this?

**Pull request closed without merging**